### DISCUSSION TITLE
Login with device doesn't work, PUT response status 422 - Unprocessable entity

### DISCUSSION THREAD

**Discussion by Majestic7979** [dani-garcia/vaultwarden#4032] [Q&A]
I am running behind traefik 2, I get the websockets connection just fine. I also get push notifications. I get the fingerprint phrase popup on my cellphone and in my desktop application.

Doing F12 on the browser and the Bitwarden desktop to see what's going on...

GET status 200 response:
```
{
    "creationDate": "2023-11-05T17:19:35.974556178Z",
    "id": "xxxxxxxxxxxx-xxxx-xxxxx-xxxxx-xxxxxxxxx",
    "key": "",
    "masterPasswordHash": "",
    "object": "auth-request",
    "origin": "https://my.own.domain",
    "publicKey": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
    "requestApproved": null,
    "requestDeviceType": "Chrome Browser",
    "requestIpAddress": "192.168.0.1",
    "responseDate": null
}
```

PUT status 422 payload:
```
{"key":"xxxxxxxxxxxxxxxxxxxxxxxxx","deviceIdentifier":"xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxx","requestApproved":true}
```
Response
```
{
  "error": {
    "code": 422,
    "reason": "Unprocessable Entity",
    "description": "The request was well-formed but was unable to be followed due to semantic errors."
  }
}
```

I am also a little confused as to why my requestIpAddress is my router's IP on my local network!? If I'm running from traefik shouldn't my IP be the public internet IP at the WAN?

End result is that I receive the notification but when I approve or deny nothing happens. On the cellphone I get an additional error: "We were unable to process your request. Please try again or contact us".

**Comment by BlackDex** [comment]
You are probably not running the `testing` tagged version as is mentioned in all other issues/discussions which report this.
Please check which version you are using and make sure it is `testing`, and also make sure you are using the new websocket proxy configuration which are provided at our wiki: https://github.com/dani-garcia/vaultwarden/wiki/Proxy-examples

**Comment by Majestic7979** [comment]
> You are probably not running the `testing` tagged version as is mentioned in all other issues/discussions which report this. Please check which version you are using and make sure it is `testing`, and also make sure you are using the new websocket proxy configuration which are provided at our wiki: https://github.com/dani-garcia/vaultwarden/wiki/Proxy-examples

Thanks, I don't think I was as clear as I could've been. I am NOT using 3012 websockets. I'm using the new Rocket based one. I can see the WSS in the console. I get instant sync between clients. This is the only issue I'm experiencing.

For reference this is how I spin up my container:

```
docker create --name=vaultwarden --restart=always -v /home/user1/dockerstore/vw-config/:/data/ -p 192.168.0.100:8088:80 vaultwarden/server:latest && docker start vaultwarden
```

So you're saying I have to spin up a container with a "testing" tag?

**Comment by BlackDex** [comment]
> So you're saying I have to spin up a container with a "testing" tag?

Yes, that is what i am saying, as is mentioned in all the other issues which reported this exact same error.

**Reply by Majestic7979** [reply]
You are coming across angry, if you are feeling angry it's really not worth your energy over this. I took time to read all the documentation to configure my environment correctly and I'm currently sick, unwell and struggling to process information, I'm doing my best and considering I've spun up traefik, authelia, and this all on my own for the past week while having a high fever I'm not gonna let it faze me. I don't think I deserve it, or anyone really. Go have a beer, de-stress yourself if you're feeling the pressure. Don't take out on people you don't know anything about. I did the "testing" tag and it's working. Despite the hostility I will say thanks for the information, it is fully functional.

**Reply by BlackDex** [reply]
Im not angry, i just don't understand that after a lot of issue and discussion threads about this, which would have shown up in a search, this question is still asked.

A simple search would have helped you solve your problem.

**Reply by Majestic7979** [reply]
I did not want to search, please get over it. I have a massive migraine, a massive fever, massive body aches, and my eyes hurt. I wanted to ask a simple question and I provided some relevant details, I was looking for a simple, friendly answer, not an attack. I'm sure you are a nice person IRL but you are stressed obviously. Take it easy, you are much better than that. Anyway thanks again for the 2-second solution that resolved the issue.

**Reply by BlackDex** [reply]
Of you are this ill, i would suggest to not work behind a computer at all.

Also, we do ask to first check our testing version and search before reporting a new issue, which you just ignored. That is not nice in my opinion.

Also, this would have been better suited for the discussion instead of an issue in the first place.

Creating a new issue and not searching first is just lazy i think.

We each are entitled to our opinions, i respect yours, i hope you also respect and understand mine that it is a bit frustrating that people just open issues which are talked about a lot of times already.

In any case, glad it is solved for you. And hopefully other people which have the same issue will find this.

**Comment by Majestic7979** [comment]
I support two people on GitHub - go on my profile and see - and the reason why I give them money every month is because they never talk like you. They simply help. They don't care if it's been asked a million times. They can link to an existing issue and that's job done without causing offence. You like creating arguments. If I am an inconvenience I won't ever come back, so let me know and I promise you you won't see me here again. I worked in technical support beginning of my IT career. People asked me the same questions over and over for 2 years, and I never lost it, never became frustrated, it was a job, I did it. This is a job too, whether you are paid or not. When you choose to answer someone's query you treat them with respect and the way you treated me was disrespectful. I thanked you for the solution but I definitely don't appreciate or think your attitude is nice. If you are frustrated with doing what you're doing, I suggest you quit (since you made the rude comment about working through my illness...) It's a fact of life that in IT users will go for the easiest path. I didn't force you to respond. Please do not extend this, I really don't want to.

**Reply by BlackDex** [reply]
Very good that you support a few people in the open source community, that is always a nice thing to do, i don't know why this is even relevant here but still kudos too you. Same btw for mentioning being ill (while i do understand that it is not fun being ill), it has no value in this discussion, it doesn't exempt you from reading or searching or being nice.

My main reason is that you seemed to just not have read anything we asked you to do when creating the issue, not the text at the report a bug button, or the pre-filled comments at the top of the input textbox. You even mentioned yourself you didn't want to search, and i find that lazy and even a bit offensive. It wouldn't even mattered probably if we had this in a documentation or faq which would provided you the answer, *you* just didn't want to search, which probably would have saved you from this whole debate 😉. 
And i find it respectful of you follow the questions and remarks of the maintainers of a project, and disrespectful if you do not, which is the road you took.

To make it clear, this is not a job, but a hobby which i like to do. I spent a lot of my spare time into this project (and others) because i like it. I don't mind if people donate too me or not for my efforts (while it is much appreciated), nice thanks or thumbs up are nice gestures too, that is the same for being treated with respect. And I'm not frustrated with _doing what I'm doing_, if that was the case i probably wouldn't even have responded to this question (which is what this was, and not an issue) at all. I find it even offensive that you dictate this to me and you suggest to leave this project just because you find my comments not nice. You don't even know me. I have been working on this project for almost 5 years without excepting anything from anyone, and i will happily do this the coming 5 years or more.

And if you have perceived my comments regarding that you are Ill and it's probably better to not be behind a computer as rude, then i apologize for that (we Dutch people tend to be a bit direct, me maybe a bit more hehe), but it was actually ment as a suggestion, since being ill takes a lot of energy, and working and not being able to focus doesn't help in your recovery. That you assumed otherwise, and i had rude intent is not something i can help.

I just hope that you do not take any open source software for granted, and your free (unpaid) support from me or anyone else who helps the Vaultwarden project (or any other free or open source project) to be what it is today as an earned right in any form.

I hope you feel a bit better soon.
Cheers.
